### DISCUSSION TITLE
Where can I find the new SPA templates?

### DISCUSSION THREAD

**Discussion by zcxsythenew** [dotnet/aspnetcore#51341] [Q&A] [answered]
Issue #49388 and PR #49413 have removed the old SPA templates, and the issue says we "recommend the new Visual Studio templates". But where are they? I have searched the issue, the PR and the [repository](/dotnet/spa-templates) but I cannot find any of them, or any link to them. Are they unavailable on GitHub?

I have found something to contribute. When I create a React + ASP.NET project enabling Docker, the backend server listens to two random ports by default, as shown below.

![image](https://github.com/dotnet/aspnetcore/assets/30565051/645d36e3-2edc-4d68-bf81-4e52ab0ba1a5)

But the frontend uses port 7291, so it cannot receive any data from backend.

![image](https://github.com/dotnet/aspnetcore/assets/30565051/56581a4b-127e-4768-b023-becf7b1c2cba)

To fix this issue, we can modify `launchSettings.json`, remove `publishAllPorts` and use `httpPort` and `sslPort` instead. We also need to use ports `80` and `443` instead of `8080` and `8081`.

![image](https://github.com/dotnet/aspnetcore/assets/30565051/07bb5ee2-2197-4c99-a18e-96605ef5fceb)

However, where can I create such an issue?

**Comment by danroth27** [comment, marked as answer]
@zcxsythenew The new SPA templates in VS aren't currently open source, but you can raise issues about them by submitting Visual Studio feedback: Help > Send Feedback.
